src/store/reducers/loginReducer.ts

```diff
@@ -1,17 +1,54 @@
-import { LoginActions, loginToggle } from '../actions/login';
+import {
+  LoginActions,
+  loginToggle,
+  saveUsers,
+  setCurrentUser,
+} from '../actions/login';
+
+export type UserType = {
+  userEmail?: string;
+  userId: string;
+  userName: string;
+};
+
+export type initialStateType = {
+  currentUser: {
+    userEmail?: string;
+    userId: string;
+    userName: string;
+  };
+  isLoggedIn: boolean;
+  users: UserType[];
+};
 
 const initialState = {
   isLoggedIn: false,
+  users: [] as UserType[],
+  currentUser: {} as UserType,
```

Comment:
the another way to avoid `as`, is to createa a default object with with initial values